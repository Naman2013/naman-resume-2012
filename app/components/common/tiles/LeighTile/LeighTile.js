import React from 'react';
import PropTypes from 'prop-types';
import style from './LeighTile.style';

const LeighTile = ({ title, anchorText, createLink }) => (
  <div className="root">
    <h5 className="title">{title}</h5>
    <div className="action-container">
      <a className="action" href={createLink}>{anchorText}</a>
    </div>
    <style jsx>{style}</style>
  </div>
);

LeighTile.propTypes = {
  title: PropTypes.string.isRequired,
  anchorText: PropTypes.string.isRequired,
  createLink: PropTypes.func,
};

LeighTile.defaultProps = {
  createLink: () => '#',
};

export default LeighTile;
