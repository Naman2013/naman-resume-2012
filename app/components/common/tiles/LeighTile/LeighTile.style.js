import css from 'styled-jsx/css';
import { primaryFont, secondaryFont } from 'app/styles/variables/fonts';
import { geyser } from 'app/styles/variables/colors_tiles_v4';
import { screenMedium } from 'app/styles/variables/breakpoints';


export default css`
  .root {
    margin: 0 auto;
    padding: 0;
    padding: 40px;
    box-shadow: 0px 0px 8px 1px rgba(65,86,113,.2);
    width: 96%;
  }

  .title {
    margin: 0;
    padding: 0;
    font-family: ${secondaryFont};
    font-weight: normal;
    font-size: 18px;
    text-transform: capitalize;
    color: #41566F;
    border-bottom: 1px solid ${geyser};
    padding-bottom: 30px;
  }

  .action-container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-top: 20px;
  }

  .action-container {
    margin: 0;
    padding: 0;
    padding-top: 20px;
    font-family: ${primaryFont};
    font-weight: 800;
    text-transform: uppercase;
    text-decoration: none;
    font-size: 10px;
    color: #41566F;
  }

  @media ${screenMedium} {
    .root {
      width: 100%;
    }
  }
`;
