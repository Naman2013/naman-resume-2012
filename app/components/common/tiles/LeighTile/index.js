import LeighTile from './LeighTile';

export default LeighTile;
