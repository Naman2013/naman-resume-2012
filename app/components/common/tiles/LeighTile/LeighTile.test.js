import React from 'react';
import { shallow } from 'enzyme';
import LeighTile from './LeighTile';

describe('LeighTile', () => {
  const shallowWrapper = shallow(
    <LeighTile title="Foo" anchorText="Topic 1" link="#" />
  );
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
