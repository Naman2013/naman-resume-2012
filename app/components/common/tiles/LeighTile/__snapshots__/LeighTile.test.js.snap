// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeighTile should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeighTile
    anchorText="Topic 1"
    createLink={[Function]}
    title="Foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="jsx-3313364838 title"
        >
          Foo
        </h5>,
        <div
          className="jsx-3313364838 action-container"
        >
          <a
            className="jsx-3313364838 action"
            href={[Function]}
          >
            Topic 1
          </a>
        </div>,
        <JSXStyle
          css={Array []}
          styleId="3313364838"
        />,
      ],
      "className": "jsx-3313364838 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foo",
          "className": "jsx-3313364838 title",
        },
        "ref": null,
        "rendered": "Foo",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="jsx-3313364838 action"
            href={[Function]}
          >
            Topic 1
          </a>,
          "className": "jsx-3313364838 action-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Topic 1",
            "className": "jsx-3313364838 action",
            "href": [Function],
          },
          "ref": null,
          "rendered": "Topic 1",
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [],
          "styleId": "3313364838",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="jsx-3313364838 title"
          >
            Foo
          </h5>,
          <div
            className="jsx-3313364838 action-container"
          >
            <a
              className="jsx-3313364838 action"
              href={[Function]}
            >
              Topic 1
            </a>
          </div>,
          <JSXStyle
            css={Array []}
            styleId="3313364838"
          />,
        ],
        "className": "jsx-3313364838 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foo",
            "className": "jsx-3313364838 title",
          },
          "ref": null,
          "rendered": "Foo",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="jsx-3313364838 action"
              href={[Function]}
            >
              Topic 1
            </a>,
            "className": "jsx-3313364838 action-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Topic 1",
              "className": "jsx-3313364838 action",
              "href": [Function],
            },
            "ref": null,
            "rendered": "Topic 1",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [],
            "styleId": "3313364838",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
